contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3505,8 +3526,9 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 						int			role_oid = get_role_oid(role_name, true);
 
 						if (!OidIsValid(role_oid) ||
-							!is_member_of_role(GetSessionUserId(), get_sysadmin_oid()) ||
-							role_oid == get_bbf_role_admin_oid())
+							(!is_member_of_role(GetSessionUserId(), get_sysadmin_oid()) && 
+							!is_member_of_role(GetSessionUserId(), securityadmin_oid)) ||
```

Comment:
We already check for membership before at line 3507

Reply:
Instead let us add check for sysadmin and restrict its drop here. Currently we rely on postgres on restricting dropping sysadmin.
I could run this on psql endpoint
```
drop owned by sysadmin;
```
and then drop sysadmin from TDS endpoint.

Reply:
we have already blocked for securityadmin and bbf_role_admin role to get dropped, will add the same for sysadmin as well.